### PR TITLE
fix: bump axios

### PR DESCRIPTION
**Pull request by leotm**
## **Description**

Fix CI failing on Improved Yarn Audit step

```console
metamask@7.10.0 /Users/leo/Documents/GitHub/metamask-mobile
├─┬ @consensys/on-ramp-sdk@1.23.0
│ └── axios@0.27.2
├─┬ appium-adb@9.14.11
│ └─┬ @appium/support@4.1.6
│   └── axios@1.4.0
├─┬ appium@1.22.3
│ ├─┬ appium-android-driver@4.54.0
│ │ ├─┬ appium-chromedriver@4.28.0
│ │ │ └── axios@0.27.2
│ │ └── axios@0.27.2
│ ├─┬ appium-base-driver@7.11.3
│ │ └── axios@0.27.2
│ ├─┬ appium-ios-driver@4.8.3
│ │ └── axios@0.27.2
│ ├─┬ appium-mac2-driver@0.14.1
│ │ └── axios@0.27.2
│ ├─┬ appium-support@2.55.0
│ │ └── axios@0.27.2
│ ├─┬ appium-uiautomator2-driver@1.75.0
│ │ └── axios@0.27.2
│ ├─┬ appium-xcuitest-driver@3.62.0
│ │ └─┬ appium-webdriveragent@3.17.0
│ │   └── axios@0.27.2
│ └── axios@0.27.2
├── axios@0.26.1
└─┬ chromedriver@99.0.0
  └── axios@0.24.0
```

## **Related issues**

- CI broken past few hrs on Improved Yarn Audit step
- https://github.com/advisories/GHSA-wf5p-g6vw-rhxx
- https://github.com/MetaMask/metamask-mobile/pull/7747 worth looking at after

## **Manual testing steps**

```bash
yarn audit:ci
```

### **Before**

```console
Vulnerability Found:

  Severity: MODERATE
  Modules: @consensys/on-ramp-sdk>axios, appium-adb>@appium/support>axios, axios
  URL: https://github.com/advisories/GHSA-wf5p-g6vw-rhxx

Vulnerability Found:

  Severity: MODERATE
  Modules: @consensys/on-ramp-sdk>axios, appium-adb>@appium/support>axios, axios
  URL: https://github.com/advisories/GHSA-wf5p-g6vw-rhxx

Vulnerability Found:

  Severity: MODERATE
  Modules: @consensys/on-ramp-sdk>axios, appium-adb>@appium/support>axios, axios
  URL: https://github.com/advisories/GHSA-wf5p-g6vw-rhxx
```

```console
Vulnerability Found:

  Severity: MODERATE
  Modules: axios
  URL: https://github.com/advisories/GHSA-wf5p-g6vw-rhxx
```

### **After**

```console
➜  metamask-mobile git:(fix/unbreak-ci-audit-fix-CVE-2023-45857) ✗ yarn audit:ci
yarn run v1.22.19
warning ../package.json: No license field
$ ./scripts/yarn-audit.sh
warning ../package.json: No license field
$ /Users/leo/Documents/GitHub/metamask-mobile/node_modules/.bin/improved-yarn-audit --ignore-dev-deps --min-severity moderate --fail-on-missing-exclusions
Improved Yarn Audit - v3.0.0

Reading excluded advisories from .iyarc
Minimum severity level to report: moderate
Excluded Advisories: ["GHSA-p8p7-x288-28g6","GHSA-c2qf-rxjj-qqgw"]

Running yarn audit...

Found 0 vulnerabilities

2 ignored because they are dev dependencies

6 ignored because of advisory exclusions

Run `yarn audit` for more information
✔ Audit shows _zero_ moderate or high severity advisories _in the production dependencies_
✨  Done in 2.32s.
```

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by leotm — COMMENTED** [review]
consider v1.6.1, spotted in
- https://github.com/MetaMask/metamask-mobile/pull/7747

https://github.com/axios/axios/releases/tag/v1.6.1

File: package.json [hunk 1]
```diff
@@ -151,7 +151,8 @@
     "**/fast-xml-parser": "4.2.4",
     "tough-cookie": "4.1.3",
     "crypto-js": "4.2.0",
-    "browserify-sign": "4.2.2"
+    "browserify-sign": "4.2.2",
+    "axios": "1.6.0"
```
Comment (leotm) [inline package.json:155]: - https://github.com/MetaMask/metamask-mobile/pull/7758/commits/337dd4172eefc970ffd30c795acc10d684fa998b

File: package.json [hunk 2]
```diff
@@ -217,7 +218,7 @@
     "@xmldom/xmldom": "^0.8.10",
     "appium-adb": "^9.11.4",
     "asyncstorage-down": "4.2.0",
-    "axios": "^0.26.1",
+    "axios": "1.6.0",
```
Comment (leotm) [inline package.json:221]: https://github.com/MetaMask/metamask-mobile/pull/7758/commits/1e0aacba62fdfdb26469ee48ef65aae581778eda

File: package.json (hunk 1 above)
Comment (leotm) [inline package.json:155]: https://github.com/MetaMask/metamask-mobile/commit/337dd4172eefc970ffd30c795acc10d684fa998b

**Comment by leotm** [comment]
@SocketSecurity ignore axios@1.6.0

network access expected

> Promise based HTTP client for the browser and node.js

File: package.json (hunk 2 above)
Comment (leotm) [inline package.json:221]: consider unpinning to
- `^1.6.0` (caret as before)
- `~1.6.0`

Comment (leotm) [inline package.json:221]: TODO: check if needs tracking this at all, super likely yes

Comment (leotm) [reply package.json:221]: yes explicitly used in few files

Screenshot 2023-11-10 at 1 21 57 pm

Comment (leotm) [reply package.json:221]: and not in our depcheckrc

**Comment by leotm** [comment]
> E2E test started on Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/9347946f-db7b-4146-8078-f5b342539d43 You can also kick off another Bitrise E2E smoke test by removing and re-applying the (Run Smoke E2E) label

✅ [ios_e2e_test](https://app.bitrise.io/build/67c5205a-ea2a-45ca-a62c-069903d3bc99)

✅ [android_e2e_test](https://app.bitrise.io/build/b828d954-d893-43cc-ae5d-e76409f21099)

File: package.json [hunk 3]
```diff
@@ -217,7 +218,7 @@
     "@xmldom/xmldom": "^0.8.10",
     "appium-adb": "^9.11.4",
     "asyncstorage-down": "4.2.0",
-    "axios": "^0.26.1",
```
Comment (leotm) [inline package.json:220]: nb: axios v1 changes https://github.com/axios/axios/releases/tag/v1.0.0

**Review by Andepande — APPROVED** [review]
LG

**Review by tommasini — APPROVED** [review]
LGTM, do we test the on ramp on the release testing @Andepande ? it would be good to check it since we are doing a major upgrade version on axios, since it is used by them

File: yarn.lock [hunk 4]
```diff
@@ -9928,37 +9928,15 @@ axios-retry@^3.1.2:
...
-    follow-redirects "^1.14.8"
-
-axios@^0.27.0, axios@^0.x:
-  version "0.27.2"
-  resolved "https://registry.yarnpkg.com/axios/-/axios-0.27.2.tgz#207658cc8621606e586c85db4b41a750e756d972"
-  integrity sha512-t+yRIyySRTp/wua5xEr+z1q60QmLq8ABsS5O9Me1AsE5dfKqgnCFzwiCZZ/cGNd1lq4/7akDWMxdhVlucjmnOQ==
-  dependencies:
-    follow-redirects "^1.14.9"
-    form-data "^4.0.0"
-
```
Comment (leotm) [inline yarn.lock:9961]: deduplication at work 💪

File: package.json (hunk 3 above)
Comment (leotm) [reply package.json:220]: nb: we have some E2E regression testing on ramp

**Comment by leotm** [comment]
merged after discussions ^ we can follow-up on comments if agreed later to
- https://github.com/MetaMask/metamask-mobile/pull/7758#discussion_r1389372620 unpin/loosen the dep/res vers cc @legobeat
- https://github.com/MetaMask/metamask-mobile/pull/7758#pullrequestreview-1724729019 bump to v1.6.1

**Pull request merged**